test/models/test_approximate_gp.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from botorch.models.approximate_gp import (
+    ApproximateGPyTorchModel,
+    SingleTaskVariationalGP,
+    _SingleTaskVariationalGP,
+)
+from botorch.models.transforms.outcome import Standardize
+from botorch.models.transforms.input import Normalize
+from botorch.posteriors import GPyTorchPosterior
+from botorch.sampling import IIDNormalSampler
+from botorch.utils.testing import BotorchTestCase
+from gpytorch.likelihoods import GaussianLikelihood, MultitaskGaussianLikelihood
+from gpytorch.mlls import VariationalELBO
+from gpytorch.variational import (
+    IndependentMultitaskVariationalStrategy,
+    VariationalStrategy,
+)
+
+
+class TestApproximateGP(BotorchTestCase):
+    def setUp(self):
+        super().setUp()
+        self.train_X = torch.rand(10, 1, device=self.device)
+        self.train_Y = torch.sin(self.train_X) + torch.randn_like(self.train_X) * 0.2
+
+    def test_initialization(self):
+        # test non batch case
+        model = ApproximateGPyTorchModel(train_X=self.train_X, train_Y=self.train_Y)
+        self.assertIsInstance(model.model, _SingleTaskVariationalGP)
+        self.assertIsInstance(model.likelihood, GaussianLikelihood)
+        self.assertIsInstance(model.model.variational_strategy, VariationalStrategy)
+        self.assertEqual(model.num_outputs, 1)
+
+        # test batch case
+        stacked_y = torch.cat((self.train_Y, self.train_Y), dim=-1)
+        model = ApproximateGPyTorchModel(
+            train_X=self.train_X, train_Y=stacked_y, num_outputs=2
+        )
+        self.assertIsInstance(model.model, _SingleTaskVariationalGP)
+        self.assertIsInstance(model.likelihood, MultitaskGaussianLikelihood)
+        self.assertIsInstance(
+            model.model.variational_strategy, IndependentMultitaskVariationalStrategy
+        )
+        self.assertEqual(model.num_outputs, 2)
+
+
+class TestSingleTaskVariationalGP(BotorchTestCase):
```

Comment:
Let's run tests with both `double` and `float`. I run into the following error while training the model with `double` in the internal version:
```
gpytorch/variational/variational_strategy.py in forward(self, x, inducing_points, inducing_values, variational_inducing_covar, **kwargs)
    114         # Compute the mean of q(f)
    115         # k_XZ K_ZZ^{-1/2} (m - K_ZZ^{-1/2} \mu_Z) + \mu_X
--> 116         predictive_mean = (interp_term.transpose(-1, -2) @ inducing_values.unsqueeze(-1)).squeeze(-1) + test_mean
    117 
    118         # Compute the covariance of q(f)
RuntimeError: expected scalar type Double but found Float
```

Reply:
Hmm... this works locally for me. What version of gpytorch are you using?

Reply:
Yeah, it works with the current version of the diff. Must've been an issue with an earlier version. Thanks for adding the tests!